src/main/scala/xiangshan/backend/MemBlock.scala

```diff
@@ -726,9 +738,9 @@ class MemBlockInlinedImp(outer: MemBlockInlined) extends LazyModuleImp(outer)
   // After the segment instruction directive starts executing, no other instructions should be executed.
   val vSegmentFlag = RegInit(false.B)
 
-  when(vSegmentUnit.io.in.fire){
+  when(GatedValidRegNext(vSegmentUnit.io.in.fire)) {
     vSegmentFlag := true.B
-  }.elsewhen(vSegmentUnit.io.uopwriteback.valid){
+  }.elsewhen(GatedValidRegNext(vSegmentUnit.io.uopwriteback.valid)) {
```

Comment:
Please let @Anzooooo review this.

Reply:
I think there should be no problem. :wink: